src/components/FormContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useEffect, useState } from "react";
+
+interface FormDisabledContextType {
+  isDisabled: boolean | undefined;
+  setDisabled: () => void;
+}
+
+export const FormDisabledContext = createContext<FormDisabledContextType>({
+  isDisabled: false,
+  setDisabled: () => {},
```

Comment:
In this scenario, I don't think this is the correct name for this function.
The function would toggle the state. Whoever is using the function wouldn't be allowed set the state for something specific.
I won't be able to do the following:
```jsx
setDisabled(true);
```
Do you think we can choose a more accurate name?
```suggestion
  toggleDisabled: () => {},
```

Reply:
Yes. Sounds good.